libsolidity/inlineasm/AsmAnalysis.cpp

```diff
@@ -419,15 +387,11 @@ bool AsmAnalyzer::checkAssignment(assembly::Identifier const& _variable, size_t
 
 	if (_valueSize != variableSize && variableSize != size_t(-1))
 	{
-		m_errors.push_back(make_shared<Error>(
-			Error::Type::TypeError,
-			"Variable size (" +
-			to_string(variableSize) +
-			") and value size (" +
-			to_string(_valueSize) +
-			") do not match.",
-			_variable.location
-		));
+		m_errorReporter.typeError(
+			_variable.location,
+			"Variable size (" + to_string(variableSize) + ") and value size (" +
+				to_string(_valueSize) + ") do not match."
```

Comment:
@roadriverrail can you also please address this? (The others were changes so this has too).